alacritty/src/event.rs

```diff
@@ -354,22 +352,13 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
         let mut env_args = env::args();
         let alacritty = env_args.next().unwrap();
 
-        // Use working directory of controlling process, or fallback to initial shell, then add it
-        // as working-directory parameter.
-        #[cfg(unix)]
-        let mut args = foreground_process_path()
-            .map(|path| vec!["--working-directory".into(), path])
-            .unwrap_or_default();
-
-        #[cfg(not(unix))]
         let mut args: Vec<PathBuf> = Vec::new();
 
-        let working_directory_set = !args.is_empty();
-
         // Reuse the arguments passed to Alacritty for the new instance.
         while let Some(arg) = env_args.next() {
-            // Drop working directory from existing parameters.
-            if working_directory_set && arg == "--working-directory" {
+            // Drop working directory from existing parameters; start_daemon()
+            // will set it to the parent alacritty instance current directory
+            if arg == "--working-directory" {
```

Comment:
What if we're on Windows and the working directory was set? Now you're just dropping it instead of passing it.

Reply:
I am new on cross-platform in Rust, I am making a lot of mistakes like this one, thanks for the help.